geom/type_polygon.go

```diff
@@ -495,3 +495,30 @@ func (p Polygon) PointOnSurface() Point {
 	pt, _ := pointOnAreaSurface(p)
 	return pt
 }
+
+// ForceCW returns the equivalent Polygon that has its exterior ring in a
+// clockwise orientation and any inner rings in a counter-clockwise
+// orientation.
+func (p Polygon) ForceCW() Polygon {
+	return p.forceOrientation(true)
+}
+
+// ForceCCW returns the equivalent Polygon that has its exterior ring in a
+// counter-clockwise orientation and any inner rings in a clockwise
+// orientation.
+func (p Polygon) ForceCCW() Polygon {
+	return p.forceOrientation(false)
+}
+
+func (p Polygon) forceOrientation(forceCW bool) Polygon {
+	orientedRings := make([]LineString, len(p.rings))
+	for i, ring := range p.rings {
+		alreadyCW := signedAreaOfLinearRing(ring, nil) < 0
```

Comment:
Can you comment about whether it'd be worthwhile checking if the entire polygon is already oriented correctly and if so return that?

Reply:
That's a good point. We would save additional allocations that way, since if the polygon is already oriented correctly then we don't need to allocate a new rings slice.

I'll add in some benchmarks as well, since this would technically fall into a performance optimisation.

Reply:
Yes, you're right, it's a performance optimisation so I think it needn't be in the same PR. Could this kind of change also apply to other geometry types?

Reply:
Yep, probably best to do this as a different change. I've made a ticket here: https://github.com/peterstace/simplefeatures/issues/241

Yep, would apply to MultiPolygon and GeometryCollection as well.